README.md

```diff
@@ -170,36 +134,150 @@ The following settings are available in the `.env` file:
+the listed repositories.
+
+`github_url` might be specified in case of an alternate url. Default is "github.com".
+
+`github_token` might be specified to use an alternate environment variable name to look for the
+token value. Default is "GITHUB_TOKEN"
+
+To crawl privates repositories, either you must use a [GitHub Application](#github-application)
+or you must generate a Personal Access Token with the "repo" scope.
+
+Note that this crawler is managed by the `crawler-legacy` container.
+
+###### Gerrit
+
+A Gerrit provider settings
+
+```YAML
+  provider:
+    gerrit_url: https://review.opendev.org
+    gerrit_repositories:
+      - openstack/nova
+      - openstack/neutron
+    # Optional settings
+    gerrit_login: monocle
+    gerrit_password: GERRIT_PASSWORD
+    gerrit_url_insecure: true
+    gerrit_prefix: opendev/
+```
+
+`gerrit_url` is mandatory and must be the url of the Gerrit provider.
+`gerrit_repositories` is mandatory and is the list of repositories from which the crawler will fetch Reviews from.
+
+`gerrit_login` might be specified to authenticate on the provider API.
+`gerrit_password`might be specified to use an alternate environment variable name to look for the
+password. Default is "GERRIT_PASSWORD"
+
+`gerrit_url_insecure` could be set to true to prevent HTTP certificate check.
+
+`gerrit_prefix` might be set to configure the crawler to prepend the repository name with a prefix.
+
+Note that this crawler is managed by the `crawler-legacy` container.
+
+###### GitLab
+
+A GitLab provider settings
+
+```YAML
+  provider:
+    gitlab_organization: redhat/centos-stream/ci-cd/zuul
+    # Optional settings
+    gitlab_repositories:
+      - jobs-config
+    gitlab_url: https://gitlab.com/api/graphql
+    gitlab_token: GITLAB_TOKEN
+```
+
+`gitlab_organization` is the only mandatory key. If `gitlab_repositories` is not specified then
+the crawler will crawl the whole organization repositories. If specified then it will crawl only
+the listed repositories.
+
+`gitlab_url` might be specified in case of an alternate url. Default is "https://gitlab.com/api/graphql".
+
+`gitlab_token` might be specified to use an alternate environment variable name to look for the
+token value. Default is "GITLAB_TOKEN"
 
-### GitHub private repositories
+Note that this crawler is managed by the `crawler` container.
 
-To let Monocle crawl and index privates repositories, either you must use a
-[GitHub Application](#github-application) or you must generate a Personal Access Token
-with the "repo" scope.
+##### TaskData
+
+Monocle provides additional crawlers to attach tasks/issues/RFEs to changes based on a
+match on `change_url`. Then, Changes can be enhanced with information about related
+tasks such as a `priority` or a `score`.
+
+However these crawlers are not fully integrated with the configuration file and the crawler
+container process. Until they are, usage might be a bit rough.
```

Comment:
Perhaps we don't have to integrate this, it's good to have a generic task-data interface that can be used freely. In a future version the github and gitlab provider will do this automatically, and there will be a dedicated bugzilla crawler, and we can still keep that generic provider.

```suggestion
Here are some example process you can run to collect task data: 
```